ecl/hql/hqlparse.cpp

```diff
@@ -2146,6 +2154,43 @@ int HqlLex::processStringLiteral(attribute & returnToken, char *CUR_TOKEN_TEXT,
     throwUnexpected();
 }
 
+
+void HqlLex::stripSlashNewline(attribute & returnToken, StringBuffer & target, size_t len, const char * text)
+{
+    target.ensureCapacity(len);
+    for (size_t i = 0; i < len; i++)
+    {
+        char cur = text[i];
+        if (cur == '\\')
+        {
+            if (i+1 < len)
+            {
+                byte next = text[i+1];
+                if (next == '\n')
+                {
+                    i++;
+                    continue;
+                }
+                if (next == '\r')
+                {
+                    i++;
+                    if ((i+1) < len && text[i+1] == '\n')
+                        i++;
+                    continue;
+                }
+            }
+            else
+            {
+                StringBuffer msg("Can not terminate a string with escape char '\\': ");
```

Comment:
Isn't \ a line joiner rather than escape character?  Also, how do you know that you are at the end of the string here?